src/plugins/utils/js-error-utils.ts

```diff
@@ -56,11 +58,39 @@ const appendErrorPrimitiveDetails = (
     rumEvent.message = error.toString();
 };
 
-const appendErrorObjectDetails = (
+const appendErrorSourceMaps = async (
+    error: globalThis.Error,
+    fetchFunction?: SourceMapsFetchFunction
+) => {
+    if (error && error.stack) {
+        try {
+            const stackFrames = await StackTrace.fromError(error, {
```

Comment:
Issue: does this need its own try catch logic to capture validation errors?

From documentation: https://github.com/stacktracejs/stacktrace.js?tab=readme-ov-file#get-stack-trace-from-an-error

```
var error = new Error('BOOM!');

StackTrace.fromError(error).then(callback).catch(errback);
//===> Promise(Array[StackFrame], Error)
``

Reply:
I added this in the case that if an error does occur from the stacktrace lib that it gets recorded.

Reply:
Suggestion: if there are validation errors from `stacktrace-js`, then the JSErrors look like they will be dropped. We may want to record the original error instead as a backup.

Reply:
This try catch is here as a precaution is case StackTrace.fromError() fails, we would want to know so the catch error is added to the incoming error's stack . Whether StackTrace.fromError() fails or not, incoming errors are never dropped, the try catch is only attempts at replacing the incoming error's stack value, it does not return anything.